src/main/java/com/notably/model/viewstate/ViewStateModel.java

```diff
@@ -1,7 +1,30 @@
 package com.notably.model.viewstate;
 
+import java.util.Optional;
+
+import javafx.beans.property.Property;
+
 /**
  * API for the ViewStateModel component.
  */
 public interface ViewStateModel extends CommandInputModel, HelpFlagModel {
+
+    /**
+     * Shows user the meaning of their input, e.g. "Create a new note".
+     * When input cannot get corrected, the "error" message will be shown in commandTextProperty.
+     * TODO: add logo.
+     * @return The meaning of the user input.
+     */
+    Property<Optional<String>> responseTextProperty();
+
+    /**
+     * Updates the responseTextProperty.
+     * @param responseText The meaning of the "command" that the user inputs.
```

Comment:
Let's rephrase this such that it also informs about the possibility of it being an error message?
I was thinking something along the lines of "A response text to the user, which could be an explanation of the typed command or an error message"

Reply:
ok!